packages/concerto-tools/lib/codegen/fromcto/csharp/csharpvisitor.js

```diff
@@ -228,6 +228,24 @@ class CSharpVisitor {
         return null;
     }
 
+    /**
+     * Visitor design pattern
+     * @param {Field} field - the object being visited
+     * @param {Object} parameters  - the parameter
+     * @return {Object} the result of visiting or null
+     * @private
+     */
+    visitScalarField(field, parameters) {
+        let fieldType;
+        let scalarField = field.getScalarField();
+        if (scalarField.getType() === 'String') {
```

Comment:
I don't think this check is really necessary. Once we know that the scalar is of type concerto.scalar.UUID we should be good to go...